kubernetes_deploy/staging/deployment.yml

```diff
@@ -55,3 +56,62 @@ spec:
             secretKeyRef:
               name: db
               key: port
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  annotations:
+    kubernetes.io/change-cause: "<to be filled in deploy_to_kubernetes script>"
+  name: laa-legal-adviser-api-queue-worker
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: laa-legal-adviser-api-queue-worker
+        env: staging
+        service_area: laa-get-access
+        service_team: cla-fala
+    spec:
+      containers:
+      - image: "<to be set by deploy_to_kubernetes>"
+        name: app
```

Comment:
Whenever I rename the container to anything other than app I get `error: unable to find container named "app"` https://circleci.com/gh/ministryofjustice/laa-legal-adviser-api/623

Is there anything enforcing that container name should be app

Reply:
https://github.com/ministryofjustice/laa-legal-adviser-api/blob/df3148ac8f4e5c16f777aa11fb290731f84769ba/.circleci/deploy_to_kubernetes#L28-L29 does (`set image app="$ECR_DEPLOY_IMAGE"`).

However, this seems to happen because it tries to apply the same thing in both documents within the single YAML file (and only the first one has "app").

It seems that we can either
- keep "app" and a single YAML file for 2 documents
- or have separate deployment files.

Reply:
If we do go with the separate deployment files does that mean we need to have the following twice 

> cat "$NAMESPACE_DIR/deployment.yml" | \
  kubectl set image app="$ECR_DEPLOY_IMAGE" --filename=/dev/stdin --local --output=yaml | \
  kubectl annotate kubernetes.io/change-cause="$CIRCLE_BUILD_URL" --filename=/dev/stdin --local --output=yaml | \
  kubectl apply --record=false \
    --filename=/dev/stdin \
    --filename="$NAMESPACE_DIR/service.yml" \
    --filename="$NAMESPACE_DIR/ingress.yml"

Reply:
Not all of it (it's probably possible to keep a single "apply" in the end) but it gets convoluted, yeah.